### PR TITLE
At/devops/working solution

### PR DESCRIPTION
**Pull request by liquidbean**

### CODE REVIEW

File: cloudflare.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
+resource "aws_acm_certificate" "main-cert" {
```
Comment (sergio-bershadsky) [inline cloudflare.tf:1]: This chould be presented on upper level. And module must return everything needed for CloudFlare in outputs

File: examples/solution/backend.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Generated by Terragrunt. Sig: nIlQXj57tbuaRZEa
+terraform {
+  backend "s3" {
+    bucket                      = "main-dev-tf-state"
+    encrypt                     = true
+    key                         = "infra/solution/terraform.tfstate"
+    kms_key_id                  = "arn:aws:kms:eu-central-1:XXXXXXXXXXXX:key/XXXXX-XXXXX-XXXX-XXXX-XXXXX"
+    dynamodb_table              = "terraform-lock-solution"
```
Comment (sergio-bershadsky) [inline examples/solution/backend.tf:8]: This should be presented in variables

File: examples/solution/locals.tf [hunk 3]
```diff
@@ -0,0 +1,6 @@
+locals {
+  namespace = "dev"
```
Comment (sergio-bershadsky) [inline examples/solution/locals.tf:2]: Should be presented in variable

File: examples/solution/providers.tf [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+provider "aws" {
+  region                      = "eu-central-1"
+  skip_metadata_api_check     = true
+  skip_region_validation      = true
+  skip_credentials_validation = true
+
+  assume_role {
+    role_arn    = "arn:aws:iam::387673871354:role/OrganizationAccountAccessRole"
```
Comment (sergio-bershadsky) [inline examples/solution/providers.tf:20]: This chould not be here

File: examples/solution/providers.tf [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+data "aws_eks_cluster" "main" {
+  name = var.aws_eks_cluster_name
+}
+
+data "aws_eks_cluster_auth" "main" {
+  name = var.aws_eks_cluster_name
+}
+
+provider "aws" {
+  region                      = "eu-central-1"
```
Comment (sergio-bershadsky) [inline examples/solution/providers.tf:14]: This should not be here

File: locals.tf [hunk 6]
```diff
@@ -0,0 +1,9 @@
+locals {
+  vpn-ip = <<EOF
+{${var.ip}}
+EOF
+  default_tags = {
+    Environment = "development"
```
Comment (sergio-bershadsky) [inline locals.tf:6]: This should not be here

File: service.tf [hunk 7]
```diff
@@ -6,13 +6,14 @@ resource "kubernetes_service" "server" {
...
     selector = {
-      "app.kubernetes.io/name" = "${var.namespace}-${var.name}-main"
+      app = var.name
     }
     port {
       name        = "main"
       protocol    = "TCP"
-      port        = 80
+      port        = var.port
       target_port = 8000
```
Comment (sergio-bershadsky) [inline service.tf:15]: Should be passed via variable

**Review by sergio-bershadsky — CHANGES_REQUESTED** [review]

**Review by HupBaHa — APPROVED** [review]

File: examples/solution/backend.tf (hunk 2 above)
Comment (liquidbean) [reply examples/solution/backend.tf:8]: Cannot insert variables in the backend module

File: cloudflare.tf (hunk 1 above)
Comment (liquidbean) [reply cloudflare.tf:1]: Do I need to cut the cloudflare module completely from the solution?

File: locals.tf (hunk 6 above)
Comment (liquidbean) [reply locals.tf:6]: where should I move this?

**Review by sergio-bershadsky — APPROVED** [review]

**Pull request merged**